packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -626,10 +625,27 @@ - (void)pause:(FLTTextureMessage*)input error:(FlutterError**)error {
 - (void)setMixWithOthers:(FLTMixWithOthersMessage*)input
                    error:(FlutterError* _Nullable __autoreleasing*)error {
   if ([input.mixWithOthers boolValue]) {
-    [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback
-                                     withOptions:AVAudioSessionCategoryOptionMixWithOthers
-                                           error:nil];
+    if ([input.ambient boolValue]) {
+      if (@available(iOS 12.0, *)) {
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient
+                                                mode:AVAudioSessionModeVoicePrompt
+                                             options:AVAudioSessionCategoryOptionMixWithOthers
+                                               error:nil];
+      } else {
+        [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient
+                                         withOptions:AVAudioSessionCategoryOptionMixWithOthers
+                                               error:nil];
+      }
+    } else {
+      [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback
+                                       withOptions:AVAudioSessionCategoryOptionMixWithOthers
+                                             error:nil];
+    }
   } else {
+    if ([input.ambient boolValue]) {
+      [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient error:nil];
+      return;
```

Comment:
Please don't mix and match if/else (the pattern everywhere else in this method) and early returns; it makes things much harder to follow.

And in general, I think this would be much easier to follow if instead of having 5 almost-identical calls, you have a single branch for calling that's based on the `@available` check, and do the rest via populating local flag variables from the arguments and then making a single call to `setCategory:withOptions:error:`.

Reply:
I fixed this